src/platforms/alexa/directives.ts

```diff
@@ -229,6 +229,94 @@ export class RenderTemplate extends AlexaDirective implements IDirective {
   }
 }
 
+export class APLTemplate extends AlexaDirective implements IDirective {
+  public static key: string = "alexaAPLTemplate";
+  public static platform: string = "alexa";
+
+  public viewPath?: string;
+  public directive?: interfaces.alexa.presentation.apl.RenderDocumentDirective;
+
+  constructor(viewPath: string | interfaces.alexa.presentation.apl.RenderDocumentDirective) {
+    super();
+
+    if (_.isString(viewPath)) {
+      this.viewPath = viewPath;
+    } else {
+      this.directive = viewPath;
+    }
+  }
+
+  public async writeToReply(
+    reply: IVoxaReply,
+    event: IVoxaEvent,
+    transition: ITransition,
+  ): Promise<void> {
+    this.validateReply(reply);
+
+    if (!_.includes(event.supportedInterfaces, "Alexa.Presentation.APL")) {
+      return;
+    }
+
+    if (this.viewPath) {
+      this.directive = await event.renderer.renderPath(this.viewPath, event);
+    }
+
+    this.addDirective(reply);
+  }
+
+  private validateReply(reply: IVoxaReply) {
+    if (reply.hasDirective("Alexa.Presentation.APL.RenderDocument")) {
```

Comment:
is this the only validation? can we send an old school render template long with this?

Reply:
We can, the last directive in the directives array will be the one being rendered. I think just changing the position in the directiveHandlers array will be enough to make the APL template the default in case there's a RenderTemplate in the same response. If the device doesn't support APL but it does support RenderTemplate then the skill will inly send and render the RenderTemplate. I'll make that change and some tests for those cases.